### ISSUE TITLE
0.81.0 type at pos stoped to work with jsx in a lot of situations

### ISSUE THREAD

**Issue by istarkov** [facebook/flow#6900]
See example

```javascript
/* @flow */

import * as React from 'react';
const fn = props => <div root={props.data} />;
fn({ data: { world: true } });
```

Type at pos on props.data stopped to work in 0.81.0
```bash
# 4 40 is cursor on `props.da|ta` where `|` cursor
yarn flow type-at-pos ./utils/tmp.js 4 40
```
returns 
```
(unknown)
```
 
But in 0.80.0 version it returns good result
```
{|world: boolean|}
```

Even typed examples like 

```
import * as React from 'react';
const fnn = (props: { data: { world: string } }) => <div root={props.data} />;
```
stopped to work and gives `unknown`.

**Comment by panagosg7** [comment]
@jbrown215 It is indeed. I'm planning to add support for JSX in the typed AST this week.

**Comment by panagosg7** [comment]
Working on a fix right now, but it didn't make the current release. Will update here when it's rolled out.

**Comment by panagosg7** [comment]
Addressed in 64601df2f89ff2057c9899a77958031261508dd9
